index/server/main.go

```diff
@@ -23,13 +24,25 @@ import (
 )
 
 const (
-	devfileName            = "devfile.yaml"
-	devfileConfigMediaType = "application/vnd.devfileio.devfile.config.v2+json"
-	devfileMediaType       = "application/vnd.devfileio.devfile.layer.v1"
-	scheme                 = "http"
-	registryService        = "localhost:5000"
+	devfileName             = "devfile.yaml"
+	devfileConfigMediaType  = "application/vnd.devfileio.devfile.config.v2+json"
+	devfileMediaType        = "application/vnd.devfileio.devfile.layer.v1"
+	devfileVSXMediaType     = "application/vnd.devfileio.vsx.layer.v1.tar"
+	devfileSVGLogoMediaType = "image/svg+xml"
+	devfilePNGLogoMediaType = "image/png"
+	devfileArchiveMediaType = "application/x-tar"
+	scheme                  = "http"
+	registryService         = "localhost:5000"
 )
 
+var mediaTypeMapping = map[string]string{
+	"devfile.yaml": devfileMediaType,
+	"vsx":          devfileVSXMediaType,
```

Comment:
Nit: we can initialize consts for the resources as well so that we can use the consts here for map initialization and the switch block below (eg. `case devfileName, "logo.svg", "logo.png", "archive.tar":`)